File: src/main/java/com/tfg/scrumweb/entity/BusinessCanvas.java
package com.tfg.scrumweb.entity;

import java.io.Serializable;

import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.OneToOne;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Entity
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class BusinessCanvas implements Serializable{
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private String keyPartners;
    private String keyActivities;
    private String keyResources;
    private String valuePropositions;
    private String customerRelationships;
    private String channels;
    private String customersSegments;
    private String costStructure;
    private String revenueStreams;
}

File: src/main/resources/static/js/proyecto/businesscanvas.js
function guardarBusinessCanvas(idBusinessCanvas){
    let keyPartners = document.getElementById("keyPartners").value;
    let keyActivities = document.getElementById("keyActivities").value;
    let keyResources = document.getElementById("keyResources").value;
    let valuePropositions = document.getElementById("valuePropositions").value;
    let customerRelationships = document.getElementById("customerRelationships").value;
    let channels = document.getElementById("channels").value;
    let customersSegments = document.getElementById("customersSegments").value;
    let costStructure = document.getElementById("costStructure").value;
    let revenueStreams = document.getElementById("revenueStreams").value;

    let data = {
        "keyPartners": keyPartners,
        "keyActivities": keyActivities,
        "keyResources": keyResources,
        "valuePropositions": valuePropositions,
        "customerRelationships": customerRelationships,
        "channels": channels,
        "customersSegments": customersSegments,
        "costStructure": costStructure,
        "revenueStreams": revenueStreams
    };
    
    console.log(data);
    enviar('/api/businesscanvas/'+idBusinessCanvas, JSON.stringify(data), 'PUT');
}